### PR TITLE
add nlp example; prune pydantic dependency

### PR DESCRIPTION
**Pull request by lkevinzc**

### CODE REVIEW

File: docs/example/pytorch.md [hunk 1]
```diff
@@ -1,18 +1,37 @@
...
+We customize the `deserialize` method of the ingress stage and the `serialize` method of the egress stage. In this way, we directly manipulate the data bytes in request data with high flexibility, and write the results into response body.
+<details>
+<summary>distil_bert_sentiment.py</summary>
+```python
+--8<-- "examples/distil_bert_server_pytorch.py"
+```
+</details>
 
-### Image Segmentation
+    curl -X POST http://127.0.0.1:8000/inference -d 'i  bought this product for many times, highly recommend'
```
Comment (kemingy) [inline docs/example/pytorch.md:37]:
```suggestion
    curl -X POST http://127.0.0.1:8000/inference -d 'i bought this product for many times, highly recommend'
```

File: mosec/server.py [hunk 2]
```diff
@@ -63,12 +57,32 @@ def __init__(
...
+    def _validate_arguments(
+        worker,
+        num,
+        max_batch_size,
+        start_method,
+    ):
+        def at_least_one_int(num, name):
+            assert isinstance(
+                num, int
+            ), f"{name} must be integer but you give {type(num)}"
```
Comment (kemingy) [inline mosec/server.py:76]:
```py
isinstance(True, int)
>>> True
```

Or maybe you can try to convert with `int(num)` and cache the exception?

The function name is a little confusing. I thought it restricts a list to have at least one integer.

File: examples/resnet50_server_pytorch.py [hunk 3]
```diff
@@ -23,6 +26,7 @@
 INFERENCE_BATCH_SIZE = 16
 
 
+# define the input and output schemas for validation
 class ImageReq(BaseModel):
     image: str  # base64 encoded
```
Comment (kemingy) [inline examples/resnet50_server_pytorch.py:31]: I feel it's not a good example to demonstrate how to use `pydantic`. Only one string is required here. Maybe we can provide some simple examples that using `msgpack`, `pydantic`, `protobuf` properly.

File: examples/resnet50_server_pytorch.py [hunk 4]
```diff
@@ -32,9 +36,26 @@ class CategoryResp(BaseModel):
...
     def forward(self, req: ImageReq) -> np.ndarray:
-        im = np.frombuffer(base64.b64decode(req.image), np.uint8)
-        im = cv2.imdecode(im, cv2.IMREAD_COLOR)[:, :, ::-1]  # bgr -> rgb
+        # Customized validation for content, raise ValidationError
+        # so that the client can get 422 as http status
+        try:
+            im = np.frombuffer(base64.b64decode(req.image), np.uint8)
+            im = cv2.imdecode(im, cv2.IMREAD_COLOR)[:, :, ::-1]  # bgr -> rgb
+        except Exception as err:
+            raise ValidationError(f"cannot decode as image data: {err}")
```
Comment (kemingy) [inline examples/resnet50_server_pytorch.py:57]: `forward` should focus on the logic, not the validation part.

File: examples/echo.py [hunk 5]
```diff
@@ -2,9 +2,8 @@
 import time
 from typing import List
 
-from pydantic import BaseModel
-
 from mosec import Server, Worker
+from mosec.errors import ValidationError
```
[the remark this answers is not in the document]
Comment (lkevinzc) [reply examples/echo.py:6]: Not sure what's the benefit of that. I feel easier to distinguish all the errors (potentially there are other errors in the future) by importing them from `mosec.errors`, and `mosec` should contain more main-stream APIs.

File: mosec/errors.py [hunk 6]
```diff
@@ -0,0 +1,2 @@
+class ValidationError(Exception):
+    pass
```
[the remark this answers is not in the document]
Comment (lkevinzc) [reply mosec/errors.py:2]: Will do.

File: mosec/server.py (hunk 2 above)
Comment (lkevinzc) [reply mosec/server.py:76]: The code is interesting...
Okay will modify this part!

File: examples/resnet50_server_pytorch.py (hunk 3 above)
Comment (lkevinzc) [reply examples/resnet50_server_pytorch.py:31]: You mean more complex data structure? Different fields with various types? How about adding fields like "request_id", "img_size"?

File: examples/resnet50_server_pytorch.py (hunk 4 above)
Comment (lkevinzc) [reply examples/resnet50_server_pytorch.py:57]: Moving it into `deserialize` part? Yea but the content validation may also be considered as part of the logic. Because the user wants to check some specific data requirements, and when they fail, the user wants to raise this error instead of 500.

Comment (kemingy) [reply examples/resnet50_server_pytorch.py:57]: Any other image preprocessing needed here? For example, resizing, cropping, normalization.

File: examples/resnet50_server_pytorch.py (hunk 3 above)
Comment (kemingy) [reply examples/resnet50_server_pytorch.py:31]: Yes. Maybe we can use `pydantic` in the XGBoost example, since it contains lots of parameters for inference.

Comment (lkevinzc) [reply examples/resnet50_server_pytorch.py:31]: Ok, then I remove it from here for now.

File: examples/resnet50_server_pytorch.py (hunk 4 above)
Comment (lkevinzc) [reply examples/resnet50_server_pytorch.py:57]: It's done after this validation.

**Review by kemingy — APPROVED** [review]
